### PR TITLE
Fix TableViewCell grouped style header color

### PR DESCRIPTION
**Pull request by laminesm**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes

A bug was filed pointing out that the `TableViewCell` grouped style looked incorrect as the background was all white instead of being light grey in the header/footer areas. To fix the issue, `TableViewHeaderFooterView` background has been reverted to clear color as it used to be in fluent 1. 

### Verification

The changes were tested on the demo app.

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| before_light | after_light |
| before_dark | after_dark |
| before_keyboard_light | after_keyboard_light |
| before_keyboard_dark | after_keyboard_dark |
| before_grouped_tvc_light | tvc_light |
| before_grouped_tvc_dark | tvc_dark |

### Pull request checklist

This PR has considered:
- [x] Light and Dark appearances
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [ ] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/1320)

### CODE REVIEW

File: ios/FluentUI/Table View/TableViewCellTokenSet.swift [hunk 1]
```diff
@@ -81,10 +81,16 @@ public class TableViewCellTokenSet: ControlTokenSet<TableViewCellTokenSet.Tokens
         super.init { token, theme in
             switch token {
             case .backgroundColor:
-                return .dynamicColor { theme.aliasTokens.colors[.background1] }
+                return .dynamicColor {
+                    DynamicColor(light: theme.aliasTokens.colors[.background1].light,
```
Comment (harrieshin) [inline ios/FluentUI/Table View/TableViewCellTokenSet.swift:85]: why do we need this change? seems like both light and dark are coming from .backgoroudn1?

Comment (harrieshin) [inline ios/FluentUI/Table View/TableViewCellTokenSet.swift:85]: is there an intended usage of background1 to still use the elevated colors?

**Comment by harrieshin** [comment]
"As part of this PR, I made changes to TableViewCell's background colors to not use elevated colors so it stays consistent with its behavior in fluent 1." i am confused by this. Fluent 1 tableviewcell colors to change when elevated. please test iPad. @sophialee0416 you would have more context?

**Comment by laminesm** [comment]
> "As part of this PR, I made changes to TableViewCell's background colors to not use elevated colors so it stays consistent with its behavior in fluent 1." i am confused by this. Fluent 1 tableviewcell colors to change when elevated. please test iPad. @sophialee0416 you would have more context?

Initially I looked at #1147 but looking at #1148 it seems like only the tableview background had elevated colors.

File: ios/FluentUI/Table View/TableViewCellTokenSet.swift (hunk 1 above)
Comment (laminesm) [reply ios/FluentUI/Table View/TableViewCellTokenSet.swift:85]: The following controls use BG1:

- `BadgeField`
- `CardNudge`
- `Notification` (primary outline)
- `PersonaButtonCarousel`
- `PersonaListView`
- `SegmentedControl`

So far only `SegmentedControl` explicitly ensures it doesn't use elevated colors. I don't know how to determine if any of these controls 'need' elevated colors.

**Comment by harrieshin** [comment]
Please add screenshot of ipad split mode with group styled tableviewcell. In main branch the cell and the background color changes

**Comment by harrieshin** [comment]
Please check the ipad keyboard selection color of the cell too

**Comment by harrieshin** [comment]
whatever you do. make sure actioncells and booleancells etc are also consistent behavior.

**Review by harrieshin — APPROVED** [review]

**Pull request merged**